### ISSUE TITLE
zero versioned updates show in `--target minor`

### ISSUE THREAD

**Issue by midgleyc** [raineorshine/npm-check-updates#1213]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates`
- [x] I am using `node >= 14.14`

This is #958, but I just hit it again with passport. It looks like that was closed with #1054, but I have a use-case that I think wouldn't be satisfied with the boolean semver flag.

When doing upgrades, I like to upgrade all the minor versions (that is, versions that shouldn't have breaking changes) at once, then go through the major versions and upgrade them one-by-one. The current behaviour puts packages with major version zero in with the "non-breaking changes" packages, which isn't right.

Would you accept a `zero-is-major` flag which I could pass to exclude zero-versioned packages?

---

## Steps to Reproduce

.ncurc:

```js

```

Dependencies:

```json
{
  "dependencies": {
    "passport": "^0.5.2"
  }
}
```

Steps:

```
ncu --target minor
```

## Current Behavior

Passport is suggested to be upgraded to 0.6.0. It is highlighted in red, though.

## Expected Behavior

As zero-versioned APIs can have major version changes on any version change, no upgrade should be suggested.

**Comment by raineorshine** [comment]
I suggest using the `--filterVersion` option to exclude major version zero:

```
ncu --filterVersion "/^[^]?(\d{2}|[^0])\./"
```

Regex explanation: 

- Start with optional `^`
- Match deps with major version of 2 or more digits
- OR match single digits that are not `0`

It's not as easy as a dedicated option, but I would prefer to not increase the surface area of the API when the functionality is already available.

**Comment by midgleyc** [comment]
Okay, thanks for the alternative!

**Comment by raineorshine** [comment]
Initial support for `--target semver` added and published in `v16.11.0`.
